modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CategoriesListRowAdapter.kt

```diff
@@ -37,3 +40,26 @@ class CategoriesListRowAdapter(val onPodcastListClick: (NetworkLoadableList) ->
         holder.itemView.setOnClickListener { onPodcastListClick(category) }
     }
 }
+class CategoriesListRowRedesignAdapter(val onPodcastListClick: (NetworkLoadableList) -> Unit, val onAllCategoriesClick: (View) -> Unit) : ListAdapter<DiscoverCategory, CategoriesListRowRedesignAdapter.CategoriesRedesignViewHolder>(CATEGORY_DIFF) {
```

Comment:
I have similar suggestions as with the `CategoriesBottomSheetAdapter` to the design of this class and its nested classes.